src/overlays/actors/ovl_Object_Kankyo/z_object_kankyo.c

```diff
@@ -32,41 +57,879 @@ const ActorInit Object_Kankyo_InitVars = {
+
+            gDPPipeSync(POLY_XLU_DISP++);
+
+            gSPDisplayList(POLY_XLU_DISP++, gEffDustDL);
+
+            gDPPipeSync(POLY_XLU_DISP++);
+        }
+
+        CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_object_kankyo.c", 1127);
+    }
+}
+
+void ObjectKankyo_Lightning(ObjectKankyo* this, GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.state != 0 && globalCtx->csCtx.npcActions[0] != NULL) {
+        switch (this->effects[0].state) {
+            case 0:
+                this->effects[0].timer = 0;
+                if (globalCtx->csCtx.npcActions[0]->action == 2) {
+                    this->effects[0].state++;
+                }
+                break;
+
+            case 1:
+                if (++this->effects[0].timer >= 7) {
+                    this->effects[0].state++;
+                }
+                break;
+
+            case 2:
+                if (globalCtx->csCtx.npcActions[0]->action == 1) {
+                    this->effects[0].state = 0;
+                }
+                break;
+        }
+    }
+}
+
+void ObjectKankyo_DrawLightning(ObjectKankyo* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 pad2;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_object_kankyo.c", 1182);
+
+    if (this->effects[0].state == 1) {
+        Matrix_Translate(globalCtx->csCtx.npcActions[0]->startPos.x,
+                         globalCtx->csCtx.npcActions[0]->startPos.y,
+                         globalCtx->csCtx.npcActions[0]->startPos.z, MTXMODE_NEW);
+        Matrix_RotateX(DEG_TO_RAD(20.0f), MTXMODE_APPLY);
+        Matrix_RotateZ(DEG_TO_RAD(20.0f), MTXMODE_APPLY);
+        Matrix_Scale(2.0f, 5.0f, 2.0f, MTXMODE_APPLY);
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 255, 128);
+        gDPSetEnvColor(POLY_XLU_DISP++, 0, 255, 255, 128);
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_object_kankyo.c", 1213), G_MTX_LOAD);
+        gSPSegment(POLY_XLU_DISP++, 0x08, SEGMENTED_TO_VIRTUAL(sEffLightningTextures[this->effects[0].timer]));
+        func_80094C50(globalCtx->state.gfxCtx);
+        gSPMatrix(POLY_XLU_DISP++, &D_01000000, G_MTX_MODELVIEW | G_MTX_NOPUSH | G_MTX_MUL);
+        gDPPipeSync(POLY_XLU_DISP++);
+        gSPDisplayList(POLY_XLU_DISP++, gEffLightningDL);
+        gDPPipeSync(POLY_XLU_DISP++);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_object_kankyo.c", 1233);
+}
+
+void ObjectKankyo_SunGraveInit(ObjectKankyo* this, GlobalContext* globalCtx) {
+    s32 objectIndex = Object_GetIndex(&globalCtx->objectCtx, OBJECT_SPOT02_OBJECTS);
+
+    if (objectIndex < 0) {
+        __assert("0", "../z_object_kankyo.c", 1251);
+    } else {
+        this->demoObjectIndex = objectIndex;
+    }
+    ObjectKankyo_SetupAction(this, ObjectKankyo_LoadSunGraveObject);
+}
+
+void ObjectKankyo_LoadSunGraveObject(ObjectKankyo* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->demoObjectIndex)) {
+        this->demoObjectLoaded = true;
```

Comment:
```suggestion
        this->requiredObjectLoaded = true;
```
maybe an `is` prefix too for making it clear it's a boolean